internal/doculint/doculint.go

```diff
@@ -363,44 +363,44 @@ func validateGenDeclVariables(reporter nolint.Reporter, expr *ast.GenDecl) {
 			}
 
 			if doc == nil {
-				reporter.Reportf(vs.Pos(), "variable %q has no comment associated with it", name)
+				report.Reportf(vs.Pos(), "variable %q has no comment associated with it", name)
 				continue
 			}
 
 			if !strings.HasPrefix(strings.TrimSpace(doc.Text()), name) {
-				reporter.Reportf(vs.Pos(), "comment for variable \"%s\" should begin with \"%s\"", name, name)
+				report.Reportf(vs.Pos(), "comment for variable \"%s\" should begin with \"%s\"", name, name)
 			}
 		}
 	}
 }
 
 // validateFuncDecl ensures that an *ast.FuncDecl upholds doculint standards by ensuring
 // it has a corresponding comment that starts with the name of the function.
-func validateFuncDecl(reporter nolint.Reporter, expr *ast.FuncDecl) {
+func validateFuncDecl(report reporter.Reporter, expr *ast.FuncDecl) {
 	if expr.Name.Name == common.FuncInit {
 		// Ignore init functions.
 		return
 	}
 
 	if expr.Doc == nil {
-		reporter.Reportf(expr.Pos(), "function \"%s\" has no comment associated with it", expr.Name.Name)
+		report.Reportf(expr.Pos(), "function \"%s\" has no comment associated with it", expr.Name.Name)
 		return
 	}
 
 	// Enforce a space after the function name.
 	if !strings.HasPrefix(strings.TrimSpace(expr.Doc.Text()), expr.Name.Name+" ") {
-		reporter.Reportf(expr.Pos(), "comment for function \"%s\" should be a sentence that starts with \"%s \"", expr.Name.Name, expr.Name.Name)
+		report.Reportf(expr.Pos(), "comment for function \"%s\" should be a sentence that starts with \"%s \"", expr.Name.Name, expr.Name.Name)
 	}
 }
 
 // validatePackageName ensures that a given package name follows the conventions that can
 // be read about here: https://blog.golang.org/package-names
-func validatePackageName(reporter nolint.Reporter, pos token.Pos, pkg string) {
+func validatePackageName(report reporter.Reporter, pos token.Pos, pkg string) {
```

Comment:
"Look, if you had to type 're' at the _start_ of your name, then 'er' at the _end_ of your name, you'd be looking to strip the suffix, too."

"... have you seriously never heard of autocomplete?"